platforms/common/include/px4_platform_common/px4_work_queue/WorkItem.hpp

```diff
@@ -68,6 +64,17 @@ class WorkItem : public IntrusiveQueueNode<WorkItem *>
 
 protected:
 
+	explicit WorkItem(const char *name, const wq_config_t &config);
+	WorkItem() = delete;
+
+	virtual ~WorkItem();
+
+protected:
+
+	friend void WorkQueue::Run();
+	virtual void RunPreamble() { _run_count++; }
```

Comment:
I don't understand this one. And why is it `virtual`?

Could you do something like?

```
void Run() {
    _run_count++;
    runImpl();
}

virtual void runImpl() = 0;
```

Reply:
It's virtual because I was toying with collecting additional stats in the ScheduledWorkItem. For example we can pull the original deadline out of the hrt_call and track latency that includes time spent in hrt, scheduling, context switch, etc.

Reply:
Yes something like the runImpl() suggestion would work, although it would be easier to keep the `Run()` virtual that all the modules implement and change the first call with the counter.

Something like `void RunWrapper()` (or a better name).

Reply:
Also good. Up to you.